netutils/webclient/webclient.c

```diff
@@ -270,11 +259,23 @@ static const char g_httpcache[]      = "Cache-Control: no-cache";
  * Private Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: free_ws
+ ****************************************************************************/
+
+static void free_ws(FAR struct wget_s *ws)
+{
+  free(ws->conn);
```

Comment:
> Previously, the implementation used `free(3)` instead of `free_ws`. Since the latter does not check for `NULL` pointers, this mean the behaviour has been now changed from the older implementation when a `NULL` pointer is given (therefore, a breaking change), with a potential risk for undefined behaviour.
> 
> _Maybe_ all calls to `free_ws` within this repository are currently used with non-NULL pointers, but I would not rely on this assumption, and instead try to follow the same behaviour as `free(3)` in this regard.

as far as i know, all uses of free_ws takes non NULL ws.